packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -200,10 +200,18 @@ class FullTimelineEventGroup {
   ///   [timeline_event_B],
   ///   [timeline_event_D],
   /// ]
-  final List<List<TimelineEvent>> eventsByRow = [];
+  final eventsByRow = <List<TimelineEvent>>[];
+
+  /// At each index in the list, this stores the last event for each row, where
+  /// last means the event has the latest end time in the row.
+  ///
+  /// The most recently added event for the row is not guaranteed to be the last
+  /// event for the row.
+  final lastEventsByRow = <TimelineEvent>[];
```

Comment:
rather than creating two lists that must have the same number of elements, it could be cleaner to create a single list with one object per element
E.g. 

```dart
class RowData {
  List<TimelineEvent> events;
  TimelineEvent lastEvent;
}
```

Reply:
done